File: 704-binarySearch.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
 const search = function(nums, target) {
    let max = nums.length - 1
    let min = 0
    
    while(min <= max) {
        let half = Math.floor((max + min) / 2)
        
        if(target == nums[half])
            return half
        else if(target < nums[half])
            max = half - 1
        else
            min = half + 1
    }
    
    return -1
};

// teste com: search([-1,0,3,5,9,12], NUMERO)

File: 704-binarySearch.c
int search(int* nums, int numsSize, int target){
    int max = numsSize - 1;
    int min = 0;

    while (min <= max) {
        int half = (int)(floor(max + min / 2));

        if(nums[half] == target)
            return half;
        else if(nums[half] > target) 
            max = half - 1;
        else 
            min = half + 1;
    }

    return -1;
}